styleguide-app/Examples/Accordion.elm

```diff
@@ -143,5 +164,13 @@ type alias State =
 
 {-| -}
 update : Msg -> State -> ( State, Cmd Msg )
-update (Toggle id toExpanded) model =
-    ( Dict.insert id toExpanded model, Cmd.none )
+update msg model =
+    case msg of
+        Toggle id toExpanded ->
+            ( Dict.insert id toExpanded model, Cmd.none )
+
+        Focus id ->
+            ( model, Task.attempt Focused (Dom.focus id) )
+
+        Focused _ ->
```

Comment:
🎨 If we're never going to use this `Result`, I'd personally rather leave it out of the data model.

In that world, instead of `Task.attempt Focused (Dom.focus id)` it would be:

```
Dom.focus id
    |> Task.onError (\_ -> Task.succeed FocusAttempted)
    |> Task.perform (\_ -> FocusAttempted)
```

…which is more verbose, but I'd pay that price for the simpler `| FocusAttempted` variant in `Msg`.

Up to you though!

Reply:
My hope is that eventually, in our non-example code, we can report when focusing on an element by id fails. I think someday, when we're fully on modal v11, that will tell us when focus traps are broken/etc. Right now it would be very noisy tho 😅